invenio_communities/assets/semantic-ui/js/invenio_communities/community/searchComponents/CommunitySelectionSearch.js

```diff
@@ -0,0 +1,252 @@
+
+    const {
+      apiConfigs: { allCommunities, myCommunities },
+      autofocus,
+      overriddenComponents,
+      config,
+      userAnonymous,
+    } = this.props;
+
+    const searchApi = new InvenioSearchApi(selectedSearchApi);
+    console.log(selectedSearchApi);
+
+    const validatedComponents = this.prefixAppID(overriddenComponents, selectedAppId);
+
+    const layoutOptions = {
+      listView: true,
+      gridView: false,
+    };
+
+    const context = {
+      selectedAppId,
+      buildUID: (element) => buildUID(element, "", selectedAppId),
+      ...config,
+    };
+
+    return (
+      <OverridableContext.Provider value={validatedComponents}>
+        <SearchConfigurationContext.Provider value={context}>
+          <ReactSearchKit
+            appName={selectedAppId}
+            urlHandlerApi={{ enabled: false }}
+            searchApi={searchApi}
+            key={selectedAppId}
+            initialQueryState={selectedInitialQueryState}
+            defaultSortingOnEmptyQueryString={{ sortBy: "bestmatch" }}
+          >
+            <Grid className="m-0 pb-0 centered" relaxed padded>
+              <Grid.Row className="p-3">
+                {/* Start burguer menu for mobile and tablet only */}
+                <Grid.Column only="mobile tablet" mobile={3} tablet={1}>
+                  <Button
+                    basic
+                    size="medium"
+                    icon="sliders"
+                    onClick={() => this.setState({ sidebarVisible: true })}
+                    aria-label={i18next.t("Filter results")}
+                    className="rel-mb-1"
+                  />
+                  {/* End burguer menu */}
+                </Grid.Column>
+                <Grid.Column
+                  mobile={13}
+                  tablet={4}
+                  computer={3}
+                  floated="right"
+                  className="text-align-right-mobile"
+                >
+                  <Menu role="tablist" className="theme-primary-menu" compact>
+                    <Menu.Item
+                      as="button"
+                      role="tab"
+                      id="all-communities-tab"
+                      aria-selected={selectedAppId === allCommunities.appId}
+                      aria-controls={allCommunities.appId}
+                      name="All"
+                      active={selectedAppId === allCommunities.appId}
+                      onClick={() =>
+                        this.setState({
+                          selectedConfig: allCommunities,
+                        })
+                      }
+                    >
+                      {i18next.t("All")}
+                    </Menu.Item>
+                    <Menu.Item
+                      as="button"
+                      role="tab"
+                      id="my-communities-tab"
+                      aria-selected={selectedAppId === myCommunities.appId}
+                      aria-controls={myCommunities.appId}
```

Comment:
The front-page could reuse this "All / My communities" button. Is it worth moving it to its own component and re-using it? @ptamarit @0einstein0

Reply:
I think that not showing the buttons to unauthenticated users (as currently done in the front page) would be better. It's more of a convenience than a crucial functionality; if a community owner happens to be logged out, they will still be able to find their community.

Reply:
That's a good point. Regarding the question  "should we make this a reusable component and use it in the frontpage", wdyt?

Reply:
Yes, right now, this is mostly the same as [CommunitySelectionSearch in invenio-rdm-records](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunitySelectionModal/CommunitySelectionSearch.js).
It could go to `react-invenio-forms` and be used in all 3 places:
* The community front page
* The subcommunities search
* The deposit form community selection